include/KLFitter/LikelihoodSingleTopAllHadronic.h

```diff
@@ -0,0 +1,280 @@
+  /* @{ */
+
+  /**
+   * The default constructor. This initializes all member attributes and calls
+   * the functions DefineModelParticles() and DefineParameters().
+   */
+  LikelihoodSingleTopAllHadronic();
+
+  /// The (defaulted) destructor.
+  ~LikelihoodSingleTopAllHadronic();
+
+  /* @} */
+
+  /// Enumerator for the fitted parameters of this likelihood.
+  enum Parameters { parBhadE,   ///< Energy of the hadronic b quark
+                    parLQ1E,    ///< Energy of the light quark 1
+                    parLQ2E,    ///< Energy of the light quark 2
+                    parTopM     ///< Mass of the top quark
+  };
+
+  /** \name Member functions (Set)  */
+  /* @{ */
+
+  /**
+   * Set a flag. If flag is true the invariant top quark mass is
+   * fixed to the pole mass.
+   * @param flag The flag.
+   */
+  void SetFlagTopMassFixed(bool flag) { m_flag_top_mass_fixed = flag; }
+
+  /**
+   * Set a flag. If flag is true, take the parameter sigma values from the
+   * transfer functions.
+   * @param flag The flag.
+   */
+  void SetFlagGetParSigmasFromTFs(bool flag) { m_flag_get_par_sigmas_from_TFs = flag; }
+
+  /* @} */
+  /** \name Member functions (BAT)  */
+  /* @{ */
+
+  /**
+   * Define the parameters of the fit. This calls BCModel::AddParameter() for
+   * all parameters in the enum #Parameters.
+   */
+  void DefineParameters() override;
+
+  /**
+   * Get initial values for the parameters. This calls
+   * GetInitialParametersWoNeutrinoPz() and retrieves the neutrino solutions
+   * via GetNeutrinoPzSolutions(). The initial values are then set accordingly.
+   * @return vector of initial values.
+   */
+  std::vector<double> GetInitialParameters() override;
+
+  /**
+   * The posterior probability definition, overloaded from BCModel.
+   * @param parameters A vector of parameters (double values).
+   * @return The logarithm of the prior probability.
+   */
+  double LogLikelihood(const std::vector <double> & parameters) override;
+
+  /**
+   * The posterior probability definition, overloaded from BCModel. Instead of
+   * the final log likelihood value as in LogLikelihood(), this returns all
+   * subcomponents.
+   * @param parameters A vector of parameters (double values).
+   * @return A vector with the components of the logarithm of the prior
+   * probability. Its components are: \n
+   *   0) TF_bhad \n
+   *   1) TF_lq1 \n
+   *   2) TF_lq2 \n
+   *   3) BW_Whad \n
+   *   4) BW_Thad
+   */
+  std::vector<double> LogLikelihoodComponents(std::vector <double> parameters) override;
+
+  /* @} */
+  /** \name Member functions (misc)  */
+  /* @{ */
+
+  /**
+   * Set the values for the missing ET x and y components and the m_et_miss_sum. This
+   * sets the internal variables #m_et_miss_x, #m_et_miss_y and #m_et_miss_sum to the
+   * given values.
+   * @param etx missing ET x component.
+   * @param ety missing ET y component.
+   * @param sumet total scalar ET.
+   * @return An error flag.
```

Comment:
```suggestion
   * For events with missing ET this function would set the x and y components. In this
   * class, the function is only implemented as a trivial function as MET is not used
   * in the likelihood.
```
Sorry, I think I copy-pasted this comment blindly from the l+jets likelihood. The description for this particular likelihood would be rather confusing, so I would suggest the above change.